### PR TITLE
Add an archetype to build functions with Kotlin

### PR DESCRIPTION
**Pull request by brunoborges**

### CODE REVIEW

File: azure-functions-kotlin-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-clean-plugin</artifactId>
+                    <version>3.1.0</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+
+        <plugins>
+            <plugin>
```
Comment (Flanker32) [inline azure-functions-kotlin-archetype/src/main/resources/archetype-resources/pom.xml]: It seems lose execution configuration for kotlin
```
            <plugin>
                <groupId>org.jetbrains.kotlin</groupId>
                <artifactId>kotlin-maven-plugin</artifactId>
                <version>1.2.71</version>
                <executions>
                    <execution>
                        <id>compile</id>
                        <phase>process-sources</phase>
                        <goals>
                            <goal>compile</goal>
                        </goals>
                        <configuration>
                            <sourceDirs>
                                <source>src/main/java</source>
                                <source>src/main/kotlin</source>
                                <source>src/main/resources</source>
                            </sourceDirs>
                        </configuration>
                    </execution>
                    <execution>
                        <id>test-compile</id>
                        <phase>process-test-sources</phase>
                        <goals>
                            <goal>test-compile</goal>
                        </goals>
                        <configuration>
                            <sourceDirs>
                                <source>src/test/java</source>
                                <source>src/test/kotlin</source>
                                <source>src/test/resources</source>
                            </sourceDirs>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

File: azure-functions-kotlin-archetype/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+`appRegion` | westus | Specifies the region of your Azure Functions, which will be used when creating the new Azure Functions.
+`resourceGroup` | java-functions-group | Specifies the resource group of your Azure Functions, which will be used when creating the new Azure Functions..
+
+## Usage
+
+### Interactive Mode
+Run below command to create projects for Azure Java Functions in interactive mode.
+
+```cmd
+mvn archetype:generate -DarchetypeGroupId=com.microsoft.azure -DarchetypeArtifactId=azure-functions-archetype
```
Comment (Flanker32) [inline azure-functions-kotlin-archetype/README.md]: -DarchetypeArtifactId=azure-functions-kotlin-archetype

Comment (brunoborges) [reply azure-functions-kotlin-archetype/README.md]: Resolved in e666fef

File: azure-functions-kotlin-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (brunoborges) [reply azure-functions-kotlin-archetype/src/main/resources/archetype-resources/pom.xml]: Resolved in 5a69618

**Comment by Flanker32** [comment]
LGTM

File: azure-functions-kotlin-archetype/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Maven Archetypes for Azure Functions
+[![Maven Central](https://img.shields.io/maven-central/v/com.microsoft.azure/azure-functions-kotlin-archetype.svg)](http://search.maven.org/#search%7Cga%7C1%7Cg%3A%22com.microsoft.azure%22%20AND%20a%3A%22azure-functions-kotlin-archetype%22)
+
+This is a Maven Archetype for Azure Functions written in Kotlin.
+
+## Required Parameters
+
+Like any other Maven Archetype, you are required to provide values for parameters `groupId`, `artifactId`, `version` and `package`.
+On top of that, two extra parameters are required for Azure Functions Archetype as listed in below table. They both have default values.
```
Comment (andxu) [inline azure-functions-kotlin-archetype/README.md]: two => three

Comment (andxu) [reply azure-functions-kotlin-archetype/README.md]: both => all

Comment (brunoborges) [reply azure-functions-kotlin-archetype/README.md]: Fixed in 4146eb6

**Comment by brunoborges** [comment]
ping @andxu ?

**Comment by brunoborges** [comment]
@Flanker32 any update?

**Comment by brunoborges** [comment]
@Flanker32 @jdneo @andxu do you have an ETA for when this archetype will me merged?

**Pull request merged**